### PR TITLE
Add query args as attributes

### PR DESCRIPTION
**Pull request by jhchabran**
TODO review and send upstream PR

This PR adds support for decorating spans with the query arguments as attributes, in the form of `db.args.$n: value`. A `Skip` option is also provided to disable the mechanism if needed.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -137,9 +152,23 @@ func newConfig(options ...Option) config {
...
+func withDBStatement(ctx context.Context, cfg config, query string, args []driver.NamedValue) []attribute.KeyValue {
 	if cfg.SpanOptions.DisableQuery {
 		return cfg.Attributes
 	}
-	return append(cfg.Attributes, semconv.DBStatementKey.String(query))
+	attrs := append(cfg.Attributes, semconv.DBStatementKey.String(query))
+	if cfg.SpanOptions.ArgumentOptions.EnableAttributes {
+		opts := cfg.SpanOptions.ArgumentOptions
+		list := namedToInterface(args)
+		if opts.Skip == nil || (opts.Skip != nil && !opts.Skip(ctx, query, list)) {
```
Comment (burmudar) [inline config.go]: should we invert this if ? It's a little nicer to read
```go
// skip adding of db attributes if true
if opts.Skip != nil && opts.Skip(ctx, query, list) {
  	attrs = append(attrs, attribute.Bool("db.args.skipped", true))
} else {
        // add all the attributes
}
```

**Review by burmudar — APPROVED** [review]
Minor comment but LGTM!

File: config.go (hunk 1 above)
Comment (jhchabran) [reply config.go]: Good point, that's nicer to read this way, thanks!

File: config.go [hunk 2]
```diff
@@ -99,6 +101,19 @@ type SpanOptions struct {
 
 	// OmitConnectorConnect if set to true will suppress sql.connector.connect spans
 	OmitConnectorConnect bool
+
+	// ArgumentOptions holds configuration for query arguments.
```
Comment (bobheadxi) [inline config.go]:
```suggestion
	// ArgumentOptions holds configuration for including query arguments in spans.
```

**Review by bobheadxi — APPROVED** [review]

**Comment by jhchabran** [comment]
PR: https://github.com/XSAM/otelsql/pull/116

**Pull request closed without merging**